### PR TITLE
Adds support for arrays

### PR DESCRIPTION
**Pull request by jjacobskind**
## Problem

The VARIANT_SKU_MAPPING model has array fields that cannot currently be tapped.

## Proposed changes

Adds support for arrays

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] Description above provides context of the change
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [ ] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions

### CODE REVIEW

**Comment by cwatkins989** [comment]
This is off to a great start! Two things to hit and I think this is a quick add in the books 📚 

1) This method [here](https://github.com/bonobos/tap-snowflake/blob/develop/tap_snowflake/sync_strategies/common.py#L167) is the handler for serializing output rows from Snowflake to Singer records. Without looking at the data, _maybe_ the else logic [here](https://github.com/bonobos/tap-snowflake/blob/develop/tap_snowflake/sync_strategies/common.py#L208-L209) actually works out just fine ( as you may have mentioned via Slack) otherwise you can write logic to handle when `type == array`.

2) Testing - tests can be placed [here](https://github.com/bonobos/tap-snowflake/blob/develop/tests/integration/test_tap_snowflake.py) for now. At the moment, we have one enormous class that does the setup and then we assert a Unit Test that the serialization works (e.g, [Object Type](https://github.com/bonobos/tap-snowflake/blob/develop/tests/integration/test_tap_snowflake.py#L243-L249)) at the individual data type level + one Medium test that runs the above mentioned method to ensure all supported data types map correctly [here](https://github.com/bonobos/tap-snowflake/blob/develop/tests/integration/test_tap_snowflake.py#L251).

File: tap_snowflake/__init__.py [hunk 1]
```diff
@@ -93,12 +94,12 @@ def schema_for_column(c):
...
         result.format = 'time'
 
-    elif data_type in BINARY_TYPE:
+    elif data_type in BINARY_TYPES:
         result.type = ['null', 'string']
         result.format = 'binary'
 
-    elif data_type == 'object':
-        result.type = ['null', 'object']
+    elif data_type in OBJECT_TYPES:
```
Comment (jjacobskind) [inline tap_snowflake/__init__.py:101]: 👍

**Review by jjacobskind — COMMENTED** [review]
Looks good! The improvements you made on my stuff make sense 👍

Github won't let me approve a PR I opened, but I approve

**Review by pnadolny13 — APPROVED** [review]
Good stuff.

A random note - I dont think we have any cases of this but something thats good to be aware of: this tap grabs the schema on the fly and previously arrays were skipped but now arrays are passed through so if we had any syncs with skipped arrays they might error or have unexpected results after this is deployed since the target will receive extra columns.

**Pull request merged**